README.md

```
@@ -153,4 +153,161 @@ To verify invariants using QuickCheck:
+![image](https://github.com/user-attachments/assets/4793940e-518a-4893-b3f2-1300d488bf65)
+
+Follow these steps to prepare your environment:
+
+```bash
+# 1. Enter the Nix shell (requires Nix installed)
+nix-shell
+
+# 2. Update Cabal package index
+cabal update
+
+# 3. Ensure project dependencies are available
```

markdownlint-cli2 (0.18.1)

Table pipe style
Expected: leading_and_trailing; Actual: no_leading_or_trailing; Missing leading pipe

(MD055, table-pipe-style)